protobuf/endpoints/batch_create_artifact.proto

```diff
@@ -2,9 +2,17 @@ syntax = "proto3";
 
 package endpoints;
 
+enum ArtifactType {
+  // TODO(hayden@preemo.io, 04/17/2023): should these be ARTFACT_TYPE... or ARTIFACTTYPE_...
+  ARTIFACT_TYPE_UNSPECIFIED = 0;
+  ARTIFACT_TYPE_PARAMS = 1;
+  ARTIFACT_TYPE_RESULT = 2;
+}
+
 // This config is included for potential future use.
 // In the future it may include part count, expiration, or other metadata.
 message CreateArtifactConfig {
+  ArtifactType type = 1;
```

Comment:
we have a convention that all fields are marked optional

Reply:
got bamboozeld by the maps not being optional, but i guess protobuf isn't a fan of that?

Reply:
`map` and `repeated` cannot be `optional` because they can be empty

Reply:
i guess that makes sense